src/core/resource.cpp

```diff
@@ -277,6 +284,19 @@ EMGMResponse CResource::writeValue(forte::core::TNameIdentifier &paNameList, con
   return retVal;
 }
 
+void CResource::setInitialValues() {
+  CFunctionBlock::setInitialValues();
+  if(mInitList.empty()){
+    DEVLOG_ERROR("Error initializing values during reset\r\n");
+  }
+
+  for(std::vector<InitValue>::iterator it = mInitList.begin(); it < mInitList.end(); it++){
```

Comment:
you could use "auto it" here, and let the compiler fill in the concrete type

Reply:
and also use a [range-based for loop](https://en.cppreference.com/w/cpp/language/range-for)